clients/src/main/java/org/apache/kafka/common/utils/GeometricProgression.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.utils;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * An util class for exponential backoff, backoff, etc...
+ * The formula is Term(n) = random(1 - jitter, 1 + jitter) * scaleFactor * (ratio) ^ n
+ * If scaleFactor is greater or equal than termMax, a constant term of will be provided
+ * This class is thread-safe
+ */
+public class GeometricProgression {
+    private final int ratio;
+    private final double expMax;
+    private final long scaleFactor;
+    private final double jitter;
+
+    public GeometricProgression(long scaleFactor, int ratio, long termMax, double jitter) {
+        this.scaleFactor = scaleFactor;
+        this.ratio = ratio;
+        this.jitter = jitter;
+        this.expMax = termMax > scaleFactor ?
+                Math.log(termMax / (double) Math.max(scaleFactor, 1)) / Math.log(ratio) : 0;
+    }
+
+    public long term(long n) {
```

Comment:
nit: What about using `retries` instead of `n`? It may help to understand that the exponential value is computed based on the number of tries or retries.

Reply:
As we noticed in your earlier comments, the same value of `attempts` may correspond to different terms.
connection_timeout = constant * 2 ^ (attempts)
reconnect_backoff = constant * 2 ^ (attempts - 1)
(in KIP-580) retry_backoff = constant * 2 ^ (attempts - 1) 
So I think using `retries` or `attempts` instead of `n` might also confuse people. Shall we think of another naming?